src/main/java/jarvis/model/UniqueTaskList.java

```diff
@@ -13,17 +14,34 @@
 
 /**
  * A list of tasks that enforces uniqueness between its elements and does not allow nulls.
- * A task is considered unique by comparing using {@code Person#isSameTask(Task)}. As such, adding and updating of
- * tasks uses Task#isSameTask(Task) for equality so as to ensure that the task being added or updated is
- * unique in terms of identity in the UniqueTaskList. However, the removal of a task uses Task#equals(Object) so
- * as to ensure that the task with exactly the same fields will be removed.
+ * A task is considered unique by comparing using {@code Task#equals(Object)}.
+ * As such, adding and updating of tasks uses Task#equals(Object) for equality so as to ensure that the
+ * task being added or updated is unique in terms of identity in the UniqueTaskList. The removal of a
+ * task also uses Task#equals(Object).
  *
  * Supports a minimal set of list operations.
  *
- * @see Task#isSameTask(Task)
+ * @see Task#equals(Object)
  */
 public class UniqueTaskList implements Iterable<Task> {
 
+    private static final Comparator<Task> TASK_COMPARATOR = (t1, t2) -> {
+        if (t1.isDone() == t2.isDone()) {
+            if (t1.hasDeadline() && t2.hasDeadline()) {
+                return t1.isDone() ? t2.getDeadline().compareTo(t1.getDeadline())
+                                   : t1.getDeadline().compareTo(t2.getDeadline());
+            } else if (t1.hasDeadline()) {
+                return -1;
+            } else if (t2.hasDeadline()) {
+                return 1;
+            } else {
+                return 0;
+            }
+        } else {
+            return t1.isDone() ? 1 : -1;
+        }
```

Comment:
```suggestion
        if (t1.isDone() != t2.isDone()) {
            return t1.isDone() ? 1 : -1;
        }
        if (t1.ishasDeadline() != t2.hasDeadline()) {
            return t1.hasDeadline() ? -1 : 1;
        }
        if (!(t1.ishasDeadline() || t2.hasDeadline())) {
            return 0;
        }
        if (t1.isDone()) {
            return t2.getDeadline().compareTo(t1.getDeadline());
        } else {
            return t1.getDeadline().compareTo(t2.getDeadline());
        }
```
Reduce arrow-head code and long ternary expression